horovod/spark/common/util.py

```diff
@@ -0,0 +1,530 @@
+                         'max_size': col_max_sizes[col]}
+
+    return metadata
+
+
+def to_petastorm_fn(schema_cols, metadata):
+    ARRAY = constants.ARRAY
+    CUSTOM_SPARSE = constants.CUSTOM_SPARSE
+
+    # Convert Spark Vectors into arrays so Petastorm can read them
+    def to_petastorm(row):
+        import numpy as np
+        from pyspark import Row
+
+        fields = row.asDict().copy()
+        for col in schema_cols:
+            col_data = row[col]
+            intermediate_format = metadata[col]['intermediate_format']
+            if intermediate_format == ARRAY:
+                fields[col] = col_data.toArray().tolist()
+            elif intermediate_format == CUSTOM_SPARSE:
+                # Currently petastorm does not support reading pyspark sparse vector. We put
+                # the indices and values into one array. when consuming the data, we re-create
+                # the vector from this format.
+                size = len(col_data.indices)
+                padding_zeros = 2 * (metadata[col]['max_size'] - len(col_data.indices))
+
+                fields[col] = np.concatenate(
+                    (np.array([size]), col_data.indices, col_data.values,
+                     np.zeros(padding_zeros))).tolist()
+
+        return Row(**fields)
+
+    return to_petastorm
+
+
+def get_simple_meta_from_parquet(store, label_columns, feature_columns, sample_weight_col):
+    train_data_path = store.get_train_data_path()
+    validation_data_path = store.get_val_data_path()
+
+    if not store.exists(train_data_path):
+        raise ValueError("{} path does not exist on the store".format(train_data_path))
+
+    train_data = pq.ParquetDataset(train_data_path, filesystem=store.get_filesystem())
+    schema = train_data.schema.to_arrow_schema()
+
+    train_rows = 0
+    total_byte_size = 0
+    for piece in train_data.pieces:
+        metadata = piece.get_metadata()
+        train_rows += metadata.num_rows
+
+        pfile = piece.open()
+        file_metadata = pfile.metadata
+        for row_group_index in range(file_metadata.num_row_groups):
+            row_group = file_metadata.row_group(row_group_index)
+            total_byte_size += row_group.total_byte_size
+
+    if train_rows == 0:
+        raise ValueError('No rows found in training dataset: {}'.format(train_data_path))
+
+    if total_byte_size == 0:
+        raise ValueError('No data found in training dataset: {}'.format(train_data_path))
+
+    if train_rows > total_byte_size:
+        raise ValueError('Found {} bytes in {} rows.  Training dataset may be corrupted.'
+                         .format(total_byte_size, train_rows))
+
+    val_rows = 0
+    if store.exists(validation_data_path):
+        val_data = pq.ParquetDataset(validation_data_path, filesystem=store.get_filesystem())
+        for piece in val_data.pieces:
+            metadata = piece.get_metadata()
+            val_rows += metadata.num_rows
```

Comment:
Do we need to check the validation data as the train data?

Reply:
Good idea.  Added additional checks for the validation data.